Java SDK 4.0 Docs update (#2833)
getsentry/sentry-docs: src/platforms/java/common/configuration/transport.mdx

---
title: Transport
sidebar_order: 11
description: "Learn how to create and configure alternative HTTP transport implementation"
---

> Sentry Java SDK by default comes with a lightweight transport implementation based on [HttpUrlConnection](https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html). While this implementation is a great fit for the error reporting feature, it may provide enough throughput for high traffic, server applications using the Performance feature.

Reviewer: ```suggestion
Sentry Java SDK by default comes with a lightweight transport implementation based on [HttpUrlConnection](https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html). While this implementation is a great fit for the error reporting feature, it may provide enough throughput for high traffic, server applications using Sentry's Performance Monitoring.
```

> In a separate module, as an optional dependency SDK provides high-throughput implementation based on Apache HTTP Client 5. To use it, include following dependency in your project:

Reviewer: ```suggestion
In a separate module, as an optional dependency, the SDK also provides high-throughput implementation based on Apache HTTP Client 5. To use this module, include following dependency in your project:
```

```xml {tabTitle:Maven}
<dependency>
    <groupId>io.sentry</groupId>
    <artifactId>sentry-apache-http-client-5</artifactId>
    <version>4.0.0</version>
</dependency>
```

```groovy {tabTitle:Gradle}
implementation 'io.sentry:sentry-apache-http-client-5:4.0.0'
```

```scala {tabTitle:SBT}
libraryDependencies += "io.sentry" % "sentry-apache-http-client-5" % "4.0.0"
```

> In the next step, configure SDK to use the new transport:

Reviewer: ```suggestion
Then configure SDK to use the new transport:
```

```java {tabTitle:Java}
import io.sentry.transport.apache.ApacheHttpClientTransportFactory;

Sentry.init(options -> {
    options.setTransportFactory(new ApacheHttpClientTransportFactory());
});
```

```kotlin {tabTitle:Kotlin}
import io.sentry.transport.apache.ApacheHttpClientTransportFactory

Sentry.init {
    it.transportFactory = ApacheHttpClientTransportFactory()

> });

Reviewer 2: missing imports for `Sentry`, the idea is that ctrl c ctrl v works ootb
